File: backend/internal/constants/enums.go
package constants

type PondStatus string

const (
	PondStatusActive     PondStatus = "active"
	PondStatusQuarantine PondStatus = "quarantine"
	PondStatusClosed     PondStatus = "closed"
)

func (s ExecutionStatus) Valid() bool {
	return s == ExecutionScheduled || s == ExecutionRunning || s == ExecutionCompleted || s == ExecutionCancelled
}

func (s ExecutionStatus) CanTransitionTo(next ExecutionStatus) bool {
	switch s {
	case ExecutionScheduled:
		return next == ExecutionScheduled || next == ExecutionRunning || next == ExecutionCompleted || next == ExecutionCancelled
	case ExecutionRunning:
		return next == ExecutionRunning || next == ExecutionCompleted || next == ExecutionCancelled
	default:
		return false
	}
}

func (s PondStatus) Valid() bool {
	return s == PondStatusActive || s == PondStatusQuarantine || s == PondStatusClosed
}

type PlanStatus string

const (
	PlanStatusDraft    PlanStatus = "draft"
	PlanStatusPending  PlanStatus = "pending"
	PlanStatusApproved PlanStatus = "approved"
	PlanStatusExecuted PlanStatus = "executed"
)

func (s PlanStatus) Valid() bool {
	return s == PlanStatusDraft || s == PlanStatusPending || s == PlanStatusApproved || s == PlanStatusExecuted
}

type Role string

const (
	RoleAdmin    Role = "admin"
	RoleManager  Role = "manager"
	RoleOperator Role = "operator"
	RoleViewer   Role = "viewer"
)

func (r Role) Valid() bool {
	return r == RoleAdmin || r == RoleManager || r == RoleOperator || r == RoleViewer
}

type RiskLevel string

const (
	RiskNormal   RiskLevel = "normal"
	RiskWarning  RiskLevel = "warning"
	RiskCritical RiskLevel = "critical"
)

type ExecutionStatus string

const (
	ExecutionScheduled ExecutionStatus = "scheduled"
	ExecutionRunning   ExecutionStatus = "running"
	ExecutionCompleted ExecutionStatus = "completed"
	ExecutionCancelled ExecutionStatus = "cancelled"
)

File: frontend/src/types/enums.ts
export type PondStatus = 'active' | 'quarantine' | 'closed'
export type PlanStatus = 'draft' | 'pending' | 'approved' | 'executed'
export type RiskLevel = 'normal' | 'warning' | 'critical'
export type ExecutionStatus = 'scheduled' | 'running' | 'completed' | 'cancelled'
export type UserRole = 'admin' | 'manager' | 'operator' | 'viewer'

export const pondStatusLabels: Record<PondStatus, string> = {
  active: '运行中',
  quarantine: '隔离观察',
  closed: '已关闭',
}

export const planStatusLabels: Record<PlanStatus, string> = {
  draft: '草稿',
  pending: '待审核',
  approved: '已批准',
  executed: '已执行',
}

export const riskLabels: Record<RiskLevel, string> = {
  normal: '正常',
  warning: '预警',
  critical: '严重',
}

export const executionStatusLabels: Record<ExecutionStatus, string> = {
  scheduled: '待执行',
  running: '执行中',
  completed: '已完成',
  cancelled: '已取消',
}
